plugins/extractors/clickhouse/clickhouse.go

```diff
@@ -53,13 +55,27 @@ func New(logger log.Logger) *Extractor {
 }
 
 // Init initializes the extractor
-func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
-	if err := e.BaseExtractor.Init(ctx, config); err != nil {
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	err = e.BaseExtractor.Init(ctx, config)
+	if err != nil {
 		return err
 	}
 
-	var err error
-	e.db, err = sql.Open("clickhouse", e.config.ConnectionURL)
+	driverName := "clickhouse"
+	if config.OtelEnabled {
+		driverName, err = otelsql.Register(driverName,
+			otelsql.AllowRoot(),
+			otelsql.TraceQueryWithoutArgs(),
+			otelsql.TraceRowsClose(),
+			otelsql.TraceRowsAffected(),
+			otelsql.WithSystem(semconv.DBSystemClickhouse),
+		)
+		if err != nil {
+			return fmt.Errorf("register clickhouse otelsql wrapper: %w", err)
+		}
+	}
+
+	e.db, err = sql.Open(driverName, e.config.ConnectionURL)
```

Comment:
looks redundant, possible to extracting this out?

Reply:
updated, moved to sqlutil